products-service/handlers/get-product-by-id.ts

```diff
@@ -0,0 +1,36 @@
+import { APIGatewayProxyHandler } from 'aws-lambda';
+import * as products from '../products/products.json';
+import 'source-map-support/register';
+import { corsHeaders, errorMessages } from '../utils';
+
+export const getProductById: APIGatewayProxyHandler = async (event, _context) => {
+    try {
+        const { productId } = event.pathParameters;
+        if (productId) {
+            const product = products.find(({ id }) => productId === id);
+            if (product) {
+                return {
+                    statusCode: 200,
+                    headers: corsHeaders,
+                    body: JSON.stringify(product, null, 2),
+                };
+            }
+            return {
```

Comment:
It'd be better to throw an error here (for example, custom `NotFound` error) and handle all the errors in 1 place (error handler function).

It's a wish for the future, no need to implement it right now. It won't affect your score.

Reply:
Agreed